packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts

```diff
@@ -0,0 +1,181 @@
+    };
+
+    if (!this._isClientInitialized()) await this._initialSetup(blockSearchConfig.fromBlock);
+
+    // Fetch event information
+    // TODO: consider optimizing this further. Right now it will make a series of sequential BlueBird calls for each pool.
+    for (const l1Token of Object.keys(this.bridgePools)) {
+      const [
+        depositRelayedEvents,
+        relaySpedUpEvents,
+        relayDisputedEvents,
+        relaySettledEvents,
+      ] = await BluebirdPromise.all([
+        this.bridgePools[l1Token].getPastEvents("DepositRelayed", blockSearchConfig),
+        this.bridgePools[l1Token].getPastEvents("RelaySpedUp", blockSearchConfig),
+        this.bridgePools[l1Token].getPastEvents("RelayDisputed", blockSearchConfig),
+        this.bridgePools[l1Token].getPastEvents("RelaySettled", blockSearchConfig),
+      ]);
+
+      for (const depositRelayedEvent of depositRelayedEvents) {
+        const relayData = {
+          depositId: Number(depositRelayedEvent.returnValues.depositId),
+          sender: depositRelayedEvent.returnValues.sender,
+          slowRelayer: depositRelayedEvent.returnValues.slowRelayer,
+          disputedSlowRelayers: [],
+          instantRelayer: ZERO_ADDRESS,
+          depositTimestamp: depositRelayedEvent.returnValues.depositTimestamp,
+          recipient: depositRelayedEvent.returnValues.recipient,
+          l1Token: depositRelayedEvent.returnValues.l1Token,
+          amount: depositRelayedEvent.returnValues.amount,
+          slowRelayFeePct: depositRelayedEvent.returnValues.slowRelayFeePct,
+          instantRelayFeePct: depositRelayedEvent.returnValues.instantRelayFeePct,
+          realizedLpFeePct: depositRelayedEvent.returnValues.realizedLpFeePct,
+          priceRequestAncillaryDataHash: depositRelayedEvent.returnValues.priceRequestAncillaryDataHash,
+          depositHash: depositRelayedEvent.returnValues.depositHash,
+          depositContract: depositRelayedEvent.returnValues.depositContract,
+          relayState: relayState.Pending,
+        };
+
+        // If the local data contains this deposit ID then this is a re-relay of a disputed relay. In this case, we need
+        // to update the data accordingly as well as store the previous slow relayers.
+        if (this.relays[l1Token][relayData.depositHash]) {
+          // Bring the previous slow relayer from the data from the previous run and store it.
+          const previousSlowRelayer = this.relays[l1Token][relayData.depositHash].slowRelayer;
+          this.relays[l1Token][relayData.depositHash].disputedSlowRelayers.push(previousSlowRelayer);
+          this.relays[l1Token][relayData.depositHash].slowRelayer = relayData.slowRelayer;
+        }
+        // Else, if this if this is the first time we see this deposit hash, then store it.
+        else this.relays[l1Token][relayData.depositHash] = relayData;
+      }
+
+      // For all RelaySpedUp, set the instant relayer. Note that if a relay action was disputed the instant
```

Comment:
Did you intend to finish this sentence?

Reply:
yup. will clean it up! was part of the draft still :)